packages/jest-mock-web-client/src/index.ts

```diff
@@ -0,0 +1,338 @@
+    open: mockApi(),
+    rename: mockApi(),
+    replies: mockApi(),
+    setPurpose: mockApi(),
+    setTopic: mockApi(),
+    unarchive: mockApi(),
+  };
+
+  /**
+   * im method family
+   */
+  public readonly im = {
+    close: mockApi(),
+    history: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    open: mockApi(),
+    replies: mockApi(),
+  };
+
+  /**
+   * migration method family
+   */
+  public readonly migration = {
+    exchange: mockApi(),
+  };
+
+  /**
+   * mpim method family
+   */
+  public readonly mpim = {
+    close: mockApi(),
+    history: mockApi(),
+    list: mockApi(),
+    mark: mockApi(),
+    open: mockApi(),
+    replies: mockApi(),
+  };
+
+  /**
+   * oauth method family
+   */
+  public readonly oauth = {
+    access: mockApi(),
+    v2: {
+      access: mockApi(),
+    },
+  };
+
+  /**
+   * pins method family
+   */
+  public readonly pins = {
+    add: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * reactions method family
+   */
+  public readonly reactions = {
+    add: mockApi(),
+    get: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * reminders method family
+   */
+  public readonly reminders = {
+    add: mockApi(),
+    complete: mockApi(),
+    delete: mockApi(),
+    info: mockApi(),
+    list: mockApi(),
+  };
+
+  /**
+   * rtm method family
+   */
+  public readonly rtm = {
+    connect: mockApi(),
+    start: mockApi(),
+  };
+
+  /**
+   * search method family
+   */
+  public readonly search = {
+    all: mockApi(),
+    files: mockApi(),
+    messages: mockApi(),
+  };
+
+  /**
+   * stars method family
+   */
+  public readonly stars = {
+    add: mockApi(),
+    list: mockApi(),
+    remove: mockApi(),
+  };
+
+  /**
+   * team method family
+   */
+  public readonly team = {
+    accessLogs: mockApi(),
+    billableInfo: mockApi(),
+    info: mockApi(),
+    integrationLogs: mockApi(),
+    profile: {
+      get: mockApi(),
+    },
+  };
+
+  /**
+   * usergroups method family
+   */
+  public readonly usergroups = {
+    create: mockApi(),
+    disable: mockApi(),
+    enable: mockApi(),
+    list: mockApi(),
+    update: mockApi(),
+    users: {
+      list: mockApi(),
+      update: mockApi(),
+    },
+  };
+
+  /**
+   * users method family
+   */
+  public readonly users = {
+    conversations: mockApi(),
+    deletePhoto: mockApi(),
+    getPresence: mockApi(),
+    identity: mockApi(),
+    info: mockApi(),
+    list: mockApi(),
+    lookupByEmail: mockApi(),
+    setPhoto: mockApi(),
+    setPresence: mockApi(),
+    profile: {
+      get: mockApi(),
+      set: mockApi(),
+    },
+  };
+
+  public constructor() {
+    this.users.info.mockResolvedValue({
+      ok: true,
+      user: {
+        profile: {
+          bot_id: 'BOT_ID',
+        },
+      },
+    });
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore Typing seems to be wrong, this can take a class
```

Comment:
can we fix this?

Reply:
I'm honestly not sure if this is a bug in [jest](https://github.com/facebook/jest/blob/master/packages/jest-mock/src/index.ts#L108) or Typescript. 

I don't believe it's worth the time to figure out right now as it's fairly niche.

Reply:
(I'm choosing to spend the time trying to not have to build the mock with code vs figure this one out)